From a34bf7a25a2cfdb4426e6a363b0ebee207b3eb72 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Tue, 4 Feb 2025 18:18:12 +0100
Subject: [PATCH 1534/1695] clk: stm32mp21: remove critical clocks for M33TDCID

In M33TDCID mode don't set critical clocks (done by M33).

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 2438f6c31890352c2862d6bb91da66cfe0b3f737 clk: stm32mp21: remove critical clocks for M33TDCID]
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: I99736392a2d945eb0d4f7bf16bcf1f66cfdab8d1
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/437400
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/drivers/clk/clk-stm32mp21.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/core/drivers/clk/clk-stm32mp21.c b/core/drivers/clk/clk-stm32mp21.c
index 51d32e4fa..91e4d1e09 100644
--- a/core/drivers/clk/clk-stm32mp21.c
+++ b/core/drivers/clk/clk-stm32mp21.c
@@ -4148,8 +4148,10 @@ static void clk_stm32_set_flexgen_as_critical(void)
 	}
 }
 
-static bool clk_stm32_clock_is_critical(struct clk *clk)
+static bool clk_stm32_clock_is_critical(struct clk *clk __maybe_unused)
 {
+#ifndef CFG_STM32_CM33TDCID
+
 	struct clk *clk_criticals[] = {
 		&ck_hsi,
 		&ck_hse,
@@ -4189,7 +4191,7 @@ static bool clk_stm32_clock_is_critical(struct clk *clk)
 		if (clk == clk_critical)
 			return true;
 	}
-
+#endif /* CFG_STM32_CM33TDCID */
 	return false;
 }
 
-- 
2.47.2

